src/plugin/morphologyHDF5.cpp

```diff
@@ -213,35 +213,53 @@ void MorphologyHDF5::_readPoints(unsigned int firstSectionOffset)
                 "Reading morphology file '" + _file->getName() + "': bad number of dimensions in 'points' dataspace"));
         }
         std::vector<std::vector<float>> vec(dims[0]);
-        // points.resize(dims[0]);
         dataset.read(vec);
-        bool noNeurites = (firstSectionOffset == -1);
-        for (unsigned int i = 0; i < vec.size(); ++i) {
+
+        if (firstSectionOffset < 0) {
+            firstSectionOffset = vec.size();
+        }
+
+        somaPoints.reserve(somaPoints.size() + firstSectionOffset);
+        somaDiameters.reserve(somaDiameters.size() + firstSectionOffset);
+        for (std::size_t i = 0; i < firstSectionOffset; ++i) {
             const auto& p = vec[i];
-            if (noNeurites || i < firstSectionOffset) {
-                somaPoints.push_back({p[0], p[1], p[2]});
-                somaDiameters.push_back(p[3]);
-            } else {
-                points.push_back({p[0], p[1], p[2]});
-                diameters.push_back(p[3]);
-            }
+            somaPoints.emplace_back(Point{p[0], p[1], p[2]});
```

Comment:
Is `somaPoints` a vector of `Point`s here?
Would `emplace_back(p[0], p[1], p[2])` be ambiguous for the compiler then?

Reply:
Yes, it is due to the nature of `std::array`.